app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_setting_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(environment, content_path)
```

Comment:
How would you pass the environment to the `repository_url` from the erb template in params? It would be also unnecessary argument, as the parameter is already inside the `@host` object. You should use `@host.content_facet.lifecycle_environment` instead.